Core/src/org/sleuthkit/autopsy/contentviewers/MessageAccountPanel.java

```diff
@@ -0,0 +1,408 @@
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.LayoutStyle.ComponentPlacement;
+import javax.swing.SwingUtilities;
+import javax.swing.SwingWorker;
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle.Messages;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.centralrepository.datamodel.Persona;
+import org.sleuthkit.autopsy.centralrepository.datamodel.PersonaAccount;
+import org.sleuthkit.autopsy.centralrepository.persona.PersonaDetailsDialog;
+import org.sleuthkit.autopsy.centralrepository.persona.PersonaDetailsDialogCallback;
+import org.sleuthkit.autopsy.centralrepository.persona.PersonaDetailsMode;
+import org.sleuthkit.autopsy.centralrepository.persona.PersonaDetailsPanel;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.CommunicationsManager;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * Panel for displaying accounts and their persona information.
+ *
+ */
+final class MessageAccountPanel extends JPanel {
+
+    private static final long serialVersionUID = 1L;
+    private final static Logger logger = Logger.getLogger(MessageAccountPanel.class.getName());
+
+    private AccountFetcher currentFetcher = null;
+
+    /**
+     * Set the new artifact for the panel.
+     *
+     * @param artifact
+     *
+     * @throws TskCoreException
+     */
+    void setArtifact(BlackboardArtifact artifact) {
+        this.removeAll();
+        repaint();
+
+        if (artifact == null) {
+            return;
+        }
+
+        if (currentFetcher != null && !currentFetcher.isDone()) {
+            currentFetcher.cancel(true);
+        }
+
+        currentFetcher = new AccountFetcher(artifact);
+        currentFetcher.execute();
+    }
+
+    /**
+     * Swingworker that fetches the accounts for a given artifact
+     */
+    class AccountFetcher extends SwingWorker<List<AccountContainer>, Void> {
+
+        private final BlackboardArtifact artifact;
+
+        /**
+         * Construct a new AccountFetcher.
+         *
+         * @param artifact The artifact to get accounts for.
+         */
+        AccountFetcher(BlackboardArtifact artifact) {
+            this.artifact = artifact;
+        }
+
+        @Override
+        protected List<AccountContainer> doInBackground() throws Exception {
+            List<AccountContainer> dataList = new ArrayList<>();
+
+            CommunicationsManager commManager = Case.getCurrentCase().getSleuthkitCase().getCommunicationsManager();
+            List<Account> accountList = commManager.getAccountsRelatedToArtifact(artifact);
+            for (Account account : accountList) {
+                if (isCancelled()) {
+                    return new ArrayList<>();
+                }
+
+                Collection<PersonaAccount> personAccounts = PersonaAccount.getPersonaAccountsForAccount(account);
```

Comment:
I think some where there should be a check for CR.isEnabled(), and if so, no persona serches should be made and persona labels/buttons should be disabled...?

Reply:
I disabled the tab if Cr.isEnabled is fasle.  See MessageArtifactViewer setArtifact.